course_info/icommons.py

```diff
@@ -122,7 +121,7 @@ def _course_info_instructor_list(self, course_instance_id, **kwargs):
                 instructors = []
                 for person in rv['results']:
                     course_person_schema(person)
-                    if person.get('role')['role_id'] in INSTRUCTOR_ROLE_IDS:
+                    if person.get('role') and person.get('role')['role_id'] in INSTRUCTOR_ROLE_IDS:
```

Comment:
I was mistaken about the location of the error. However it is the same check upstream and the error happens on a similar check while sorting . When this record with null role is passed into the sort utility, 
Line 207, in views.py

course_instructor_list.sort(key=lambda x: (x.get('role', {}).get('role_id'))

ERROR	course_info.icommons:209	'NoneType' object has no attribute 'get'